js/ez-field-edit.js

```diff
@@ -0,0 +1,122 @@
+(function () {
+    /**
+     * `<ez-field-edit>` is supposed to wrap one or several HTML5 inputs. It tracks user input,
+     *  and if an input is invalid, sets its own `invalid` property to true. It will also add an
+     * `ez-invalid` class on the related label. This class will be removed once the input gets valid.
+     *
+     * Warning: each element inside <ez-field-edit> can't own more than one couple <input> + <label>
+     * Furthermore to connect a <label> to an <input> use `for`
+     * Example of a correct markup:
+     *
+     *    <ez-field-edit>
+     *        <div>
+     *            <label for="input1">First Input </label>
+     *            <input id="input1" type="text"">
+     *        <div>
+     *        <div>
+     *            <input id="input2" type="text"">
+     *            <h2>
+     *                <label for="input2">Second Input </label>
+     *            </h2>
+     *        <div>
+     *    </ez-field-edit>
+     *
+     * @polymerElement
+     * @demo demo/ez-field-edit.html
+    */
+    class FieldEdit extends Polymer.Element {
+        static get is() {
+            return 'ez-field-edit';
+        }
+
+        static get properties() {
+            return {
+                /**
+                 * Indicates if the field input is invalid.
+                 */
+                'invalid': {
+                    type: Boolean,
+                    value: false,
+                    reflectToAttribute: true,
+                },
+            };
+        }
+
+        /**
+         * Checks whether the inputs are valid or not and set the `invalid` property
+         * accordingly.
+         * Also set `ez-invalid-input` class on labels connected to invalid inputs.
+         */
+        _checkValidity() {
+            let valid;
+            const invalidInputClass = 'ez-invalid-input';
+
+            this.querySelectorAll('input, select, textarea').forEach((formElement) => {
+                const inputId = formElement.id;
+                const labels = this.querySelectorAll(`label[for="${formElement.id}"]`);
```

Comment:
it's weird to run that before checking for there's an id. I would put that a separated function. Also, you repeat the pattern twice below (condition on `valid`), so here as well, there's a room for code improvements, maybe this part could be written with something like:

```js
const getLabelsForFormElement = (input) => {
   return input.id ? this.querySelectorAll(`label[for="${formElement.id}"]`) : [];
};
const toggleInvalidClass = (valid, element) => {
    const method = valid ? 'remove' : 'add';

    element.classList[method].call(element, invalidInputClass);
}

this.querySelectorAll('input, select, textarea').forEach((formElement) => {
    const labels = getLabelsForFormElement(formElement);

    valid = formElement.validity.valid;
    if (labels.length) {
        labels.forEach(toggleInvalidClass.bind(this, valid));
    } else {
        toggleInvalidClass(formElement.parentNode, valid);
    }
});
```